test/models/product_test.rb

```diff
@@ -78,4 +78,37 @@
       expect(result).must_equal false
     end
   end
+  
+  describe "relationships" do
+    before do
+      @user = User.create(username: "test user", email: "test email") 
```

Comment:
I don't think this email will pass User's validations, so when this is run along with the User validations, the test won't pass because it won't create a User.

```suggestion
      @user = User.create(username: "test user", email: "test-email@example.com") 
```